tests/framework/microvm.py

```diff
@@ -852,6 +871,41 @@ def ssh(self):
         """Return a cached SSH connection on the 1st interface"""
         return self.ssh_iface(0)
 
+    def spawn_vhost_user_blk_backend(
+        self, host_mem_path, readonly=False, backend=VhostUserBlkBackendType.CROSVM
+    ):
+        """
+        Spawn a vhost-user-blk backend.
+
+        Return the socket path to communicate with the backend.
+        """
+        # Keep the ID short to avoid exceeding socket path length limitation (see sockaddr_un)
+        backend_id = str(uuid.uuid4())[-12:]
+
+        if backend == VhostUserBlkBackendType.QEMU:
+            backend = QemuVhostUserBlkBackend(
+                host_mem_path, self.chroot(), backend_id, readonly
+            )
+        elif backend == VhostUserBlkBackendType.CROSVM:
+            backend = CrosvmVhostUserBlkBackend(
+                host_mem_path, self.chroot(), backend_id, readonly
+            )
+        else:
+            assert False, f"unknown vhost-user-blk backend type `{backend}`"
+
+        socket = backend.spawn(self.jailer.uid, self.jailer.gid)
+        self.vhost_user_blk_backends[socket] = backend
+
+        return socket
+
+    def resize_vhost_user_drive(self, socket, new_size):
+        """Resize a vhost-user-blk drive."""
+        self.vhost_user_blk_backends[socket].resize(new_size)
+
+    def kill_vhost_user_blk_backend(self, socket):
+        """Kill a vhost-user-blk backend."""
+        del self.vhost_user_blk_backends[socket]
```

Comment:
I think both of those should be accessed as interface below:

```python
uvm.drive_vhost["drive-id"].kill()
uvm.drive_vhost["drive-id"].resize()
```

Reply:
what do you mean by `drive_vhost`? Would it only contain these two methods? Would it be unimplemented by virtio-backed drives?

Reply:
done